tests/dialects/test_arith.py

```diff
@@ -249,6 +252,63 @@ def test_select_op():
     assert select_f_op.result.type == f.result.type
 
 
+class Test_bitcast_op:
+    shape_mismatch = (
+        "'arith.bitcast' operand and result type must have compatible shape"
+    )
+    bitwidth_mismatch = "'arith.bitcast' operand and result types must have equal bitwidths or be IndexType"
+
+    @pytest.mark.parametrize(
+        "in_type, out_type",
+        [
+            (i1, IntegerType(1, signedness=Signedness.UNSIGNED)),
+            (i32, f32),
+            (i64, f64),
+            (i32, i32),
+            (IndexType(), i1),
+            (i1, IndexType()),
+            (f32, IndexType()),
+            (IndexType(), f64),
+            (VectorType(i64, [3]), VectorType(f64, [3])),
+            (VectorType(f32, [3]), VectorType(i32, [3])),
+            (MemRefType(i32, [5]), MemRefType(f32, [5])),
+        ],
+    )
+    def test_bitcast_op(self, in_type: Attribute, out_type: Attribute):
+        in_arg = TestSSAValue(in_type)
+        cast = BitcastOp(in_arg, out_type)
+
+        cast.verify_()
+        assert cast.result.type == out_type
```

Comment:
why put this in a test class?

Reply:
And below

Reply:
Ah, to use the constant strings above... I think they'd be fine at the file level also, with UPPER_SNAKE_CASE